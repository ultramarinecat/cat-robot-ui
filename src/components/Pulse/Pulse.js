//
// Pulse animation
//

import React, { memo } from 'react';

import styles from './Pulse.module.scss';
import 'load-awesome/css/ball-scale-pulse.css';

export function Pulse() {
  return (
    <div className={`${styles.main} la-ball-scale-pulse la-3x`} data-testid="pulse">
      <div />
      <div />
    </div>
  );
}

export default memo(Pulse);
