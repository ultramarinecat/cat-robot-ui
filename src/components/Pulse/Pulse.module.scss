@charset 'utf-8';

////
//// Pulse styles
////

@import 'base.scss';

.main {
  @extend %center-block;
}
