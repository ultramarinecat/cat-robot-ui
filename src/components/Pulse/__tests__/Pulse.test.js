import React from 'react';
import { render } from 'react-testing-library';

import { Pulse } from '../Pulse.js';

const PULSE = 'pulse';

describe('Pulse', () => {
  it('should display a pulse animation', () => {
    const { container, getByTestId } = render(<Pulse />);

    expect(getByTestId(PULSE)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
