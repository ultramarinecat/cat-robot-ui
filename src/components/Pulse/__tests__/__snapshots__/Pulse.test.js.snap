// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulse should display a pulse animation 1`] = `
<div
  class="main la-ball-scale-pulse la-3x"
  data-testid="pulse"
>
  <div />
  <div />
</div>
`;
